Dockerfile.build

```diff
@@ -0,0 +1,32 @@
+ARG DISTRO
+
+FROM scratch as bootstrap
+ARG CEPH_CTR_SRC=/usr/local/src/ceph
+COPY \
+    src/script/lib-build.sh \
+    src/script/run-make.sh \
+    ${CEPH_CTR_SRC}/src/script/
+COPY debian ${CEPH_CTR_SRC}/debian
+COPY \
+    ceph.spec.in \
+    do_cmake.sh \
+    install-deps.sh \
+    run-make-check.sh \
+    src/script/buildcontainer-setup.sh \
+    ${CEPH_CTR_SRC}
+
+
+FROM $DISTRO
+ENV FOR_MAKE_CHECK=1
+ARG DISTRO
+ARG CEPH_CTR_SRC=/usr/local/src/ceph
+ARG CLEAN_DNF=yes
+ARG CEPH_BRANCH=main
+COPY --from=bootstrap ${CEPH_CTR_SRC} ${CEPH_CTR_SRC}
```

Comment:
Is this line accurate?  COPY from /usr/local/src/ceph to /usr/local/src/ceph?

Reply:
Yes, the `--from=bootstrap` part means copy said path from the build container named `boostrap`. This container (layer) was already created in previous steps and populated `/usr/local/src/ceph`. (Maybe think of it more like scp'ing from host1's /usr/local/src/ceph to host2 of the same dir?)

Reply:
If you're curious why I did it this way:
* it produces fewer layers - the COPY statements in the bootstrap build container each produce a layer - but this single COPY statement produces one layer in the resulting image
* it felt nice to organize the "minimal" set of files taken from the ceph tree needed to create the build image all together in one logical block